src/llmq/ehf_signals.cpp

```diff
@@ -110,33 +117,41 @@ void CEHFSignalsHandler::HandleNewRecoveredSig(const CRecoveredSig& recoveredSig
         return;
     }
 
+    const auto ehfSignals = WITH_LOCK(cs_main, return chainstate.GetMNHFSignalsStage(chainstate.m_chain.Tip()));
     MNHFTxPayload mnhfPayload;
-    // TODO: should do this for all not-yet-signied bits
-    mnhfPayload.signal.versionBit = Params().GetConsensus().vDeployments[Consensus::DEPLOYMENT_MN_RR].bit;
-
-    const uint256 expectedId = mnhfPayload.GetRequestId();
-    LogPrint(BCLog::EHF, "CEHFSignalsHandler::HandleNewRecoveredSig expecting ID=%s received=%s\n", expectedId.ToString(), recoveredSig.getId().ToString());
-    if (recoveredSig.getId() != mnhfPayload.GetRequestId()) {
-        // there's nothing interesting for CEHFSignalsHandler
-        LogPrint(BCLog::EHF, "CEHFSignalsHandler::HandleNewRecoveredSig id is known but it's not MN_RR, expected: %s\n", mnhfPayload.GetRequestId().ToString());
-        return;
-    }
-
-    mnhfPayload.signal.quorumHash = recoveredSig.getQuorumHash();
-    mnhfPayload.signal.sig = recoveredSig.sig.Get();
-
-    CMutableTransaction tx = mnhfPayload.PrepareTx();
+    for (const auto& deployment : Params().GetConsensus().vDeployments) {
+        // skip deployments that do not use dip0023 or that have already been mined
+        if (!deployment.useEHF || ehfSignals.find(deployment.bit) != ehfSignals.end()) continue;
+
+        mnhfPayload.signal.versionBit = deployment.bit;
+        const uint256 expectedId = mnhfPayload.GetRequestId();
+        LogPrint(BCLog::EHF, "CEHFSignalsHandler::HandleNewRecoveredSig expecting ID=%s received=%s\n",
+                 expectedId.ToString(), recoveredSig.getId().ToString());
+        if (recoveredSig.getId() != mnhfPayload.GetRequestId()) {
+            // wrong deployment! Check the next one
```

Comment:
```suggestion
            // it is not known deployment or not deployment at all
```

that can be chainlock signature for example

Reply:
"wrong deployment" is referred to the current element of the loop over `Params().GetConsensus().vDeployments`, not to the `recoveredSig` that is being handled. If it creates confusion I can remove the comment